include/zephyr/net/socket.h

```diff
@@ -203,48 +222,57 @@ struct zsock_pollfd {
  *  connection ID, otherwise will contain the length of the CID value.
  */
 #define TLS_DTLS_PEER_CID_VALUE 17
-/** @} */
 
-/* Valid values for TLS_PEER_VERIFY option */
+/** Valid values for TLS_PEER_VERIFY option */
 #define TLS_PEER_VERIFY_NONE 0     /**< Peer verification disabled. */
 #define TLS_PEER_VERIFY_OPTIONAL 1 /**< Peer verification optional. */
 #define TLS_PEER_VERIFY_REQUIRED 2 /**< Peer verification required. */
 
-/* Valid values for TLS_DTLS_ROLE option */
+/** Valid values for TLS_DTLS_ROLE option */
 #define TLS_DTLS_ROLE_CLIENT 0 /**< Client role in a DTLS session. */
 #define TLS_DTLS_ROLE_SERVER 1 /**< Server role in a DTLS session. */
 
-/* Valid values for TLS_CERT_NOCOPY option */
+/** Valid values for TLS_CERT_NOCOPY option */
 #define TLS_CERT_NOCOPY_NONE 0     /**< Cert duplicated in heap */
 #define TLS_CERT_NOCOPY_OPTIONAL 1 /**< Cert not copied in heap if DER */
 
-/* Valid values for TLS_SESSION_CACHE option */
+/** Valid values for TLS_SESSION_CACHE option */
```

Comment:
might make sense to make occurences like these actual `@ref`'s

```suggestion
/** Valid values for @ref TLS_SESSION_CACHE option */
```

Reply:
I apologize as it looks like I posted several comments for the same line that had conflicting "suggested changes" :(

You can't have this be a "regular" doxygen comment -- the way it is currently it applies to `TLS_PEER_VERIFY_NONE` https://builds.zephyrproject.io/zephyr/pr/66108/docs/doxygen/html/group__secure__sockets__options.html#ga09cb746907891d86a8d69ca49717c068

Instances of such comments should either be part of a `@name`, or turned into simple `/*  ...  */` so that they are here for reference in the code itself but don't show up in the doxygen docs.

FWIW, since nesting @name is not allowed, maybe `@name Socket options for TLS` should be a `@defgroup` after all (and then added to sockets.rst as a `doxygengroup:: ...`).

Reply:
@jukkar see previous comment that you may have missed (and apologies again if my previous "suggested changes" might have been confusing).

Reply:
I indeed missed what changes I should do. And after reading your comment again and looking the generated output my wooden eyes still don't see what changes you would like to see here. Could you elaborate more?
Note that there is no nested `@name` in socket.h in this PR.